src/quotations/strategies/update-quotation/update-quotation.strategy.ts

```diff
@@ -0,0 +1,21 @@
+import { CreateQuotationDto } from '@/quotations/dtos/CreateQuotationDto';
+import { Quotation } from '@/quotations/models/quotations.model';
+import { IUpdateQuotationStrategy } from './update-quotation-strategy.interface';
+import { BadRequestException, Injectable } from '@nestjs/common';
+
+@Injectable()
+export class UpdateQuotationStrategy implements IUpdateQuotationStrategy {
+  async update(
+    quotationId: string,
+    udpateInfo: Partial<CreateQuotationDto>,
+  ): Promise<Quotation> {
+    const [affetedRows, [updateData]] = await Quotation.update(
+      { ...udpateInfo },
+      { where: { id: quotationId }, returning: true },
+    );
+    if (affetedRows === 0) {
+      throw new BadRequestException("Quotation doesn't exist");
+    }
+    return updateData.dataValues as Quotation;
+  }
+}
```

Comment:
Unhandled error for integer only total price ! Please check it carefully before pushing it

![image](https://github.com/user-attachments/assets/79932c84-e4c7-45c1-9a9a-5f3f3dae1782)